refactor(curriculum): DOM quiz questions (#58466)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/quiz-dom-manipulation-and-click-event-with-javascript/66edd07682767adff3a6231e.md

### --question--

#### --text--

What does the `innerHTML` property do?

#### --distractors--

It is a property of the `Element` used to get all text content for an HTML element.

---

It is a property of the `Element` used to add annotations to your HTML.

---

It is a property of the `Element` used to trigger events in your HTML. 

#### --answer--

It is a property of the `Element` used to set or update parts of the HTML markup.

### --question--

#### --text--

Which of the following is the correct way to use the `addEventListener()` method?

#### --distractors--

```js
btn.addEventListener("accept", () => alert("You clicked the button"));
```

---

```js
btn.addEventListener("allow", () => alert("You clicked the button"));
```

---

```js
btn.addEventListener("trigger", () => alert("You clicked the button"));
```

#### --answer--

```js
btn.addEventListener("click", () => alert("You clicked the button"));
```

### --question--

#### --text--

Which of the following is NOT a valid method?

#### --distractors--

`getElementById()`

---

`querySelector()`

---

`createElement()` 

#### --answer--

`addElementToDOM()`

### --question--

#### --text--

> Which of the following is the correct way to work with the `setAttribute()` method?

Reviewer: ```suggestion
Which of the following will set the `class` attribute to `my-class`?
```

Technically they are *all* correct, they just won't do the thing we likely wanted them to do.

What is the `Event` object?

#### --distractors--

It is a special object used to remove events from your JavaScript project.

---

It is used to automatically add events to your JavaScript projects.

---

It is a special object used when working with arrays.

#### --answer--

It is a payload that triggers when a user interacts with your web page in some way. 

### --question--

#### --text--

Which of the following is a method used to remove an event listener that was previously added to an element using the `addEventListener()` method?

#### --distractors--

`removalOfEventListener()`

---

`removedEventListener()`

---

`removingEventListener()`

#### --answer--

`removeEventListener()`

### --question--

#### --text--

What are inline event handlers?

#### --distractors--

Special attributes on an HTML element that are used to cancel all events.

---

Special attributes on an HTML element used only to trigger form submissions in your JavaScript code.

---

Special attributes on an HTML element used to remove events.

#### --answer--

Special attributes on an HTML element that are used to execute JavaScript code when an event occurs.

### --question--

#### --text--

> What are the two main categories for web API's?

Reviewer: ```suggestion
What are the two main categories for web APIs?
```

Not possessive.

#### --text--

> Which of the following is a read-only property that represents the inline style of an element?

Reviewer: ```suggestion
Which of the following is a property that represents the inline style of an element?
```

It's halfway read-only. You can't assign a style object directly, but you *can* assign a string to the property to completely overwrite existing inline styles.

#### --distractors--

`content`

---

`event`

---

`load`

#### --answer--

`style`

### --question--

#### --text--

> Which code is NOT a way to trigger a `validate` function when you click on a `button` element?

Reviewer: ```suggestion
Which example is NOT a way to trigger a `validate` function when you click on a `button` element?
```

---

```js
document.getElementById("validate_button").addEventListener("click", validate);
```

---

```js
document.getElementById("validate_button").onclick = validate;
```

#### --answer--

```js
document.getElementById("validate_button").click = validate;
```

### --question--

#### --text--

Which of the following methods is used to delay an action for a specified time?

#### --distractors--

`setInterval()`

---

`delay()`

---

`sleep()`

#### --answer--

`setTimeout()`

### --question--

#### --text--

Which of the following can be used to create and control animations directly inside JavaScript?

#### --distractors--

Allow Animations API

---

Control Animations API 

---

Create Animations API

#### --answer--

Web Animations API 

### --question--

#### --text--

What does the Canvas API do?

#### --distractors--

It is used to animate graphics in your CSS file.

---

It is used to only remove graphics from your JavaScript file.

---

It is used to remove graphics from your HTML file.

#### --answer--

> It is used to manipulate graphics right inside your JavaScript file.

Reviewer: ```suggestion
It is used to manipulate graphics via your JavaScript file.
```
